js/components/graph/Container.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Graph from "./Graph";
+import Sidebar from "./Sidebar";
+import * as focusInfo from "./sidebar/focus_descriptions";
+
+export default class Container extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currFocus: null,
+      graphName: null,
+      graphs: []
+    }
+    this.graph = React.createRef();
+    this.sidebar = React.createRef();
+  }
+
+  componentDidMount() {
+    fetch('graphs').then(res => res.json()).then(
+      (graphsData) => {
+        this.setState({
+          graphs: graphsData
+        });
+      },
+      () => {
+        throw "No graphs in database";
+      }
+    )
+    
+    // Need to use jQuery because nav-export is still a Haskell generated HTML component
+    let currGraph = this.graph.current;
+    $("#nav-export").click(function() {
+      currGraph.openExportModal();
+    });
+  }
+
+  getLocalGraph = () => {
+    // Gets graph from local storage, if it exists
+    let graphName;
+    const params = new URL(document.location).searchParams;
+    const urlSpecifiedGraph = params.get("dept");
+
+    // HACK: Temporary workaround for giving the statistics department a
+    // link to our graph.
+    // Should be replaced with a more general solution.
+    if (urlSpecifiedGraph === "sta") {
+      graphName = "Statistics";
+    } else if (urlSpecifiedGraph !== null) {
+      graphName = "Computer Science";
+    } else {
+      graphName = localStorage.getItem("active-graph") || "Computer Science";
+    }
+    this.setState({ graphName: graphName });
+    return graphName;
+  }
+
+  highlightFocus = id => {
+    if (this.graph.current.state.highlightedNodes == focusInfo[id + "FocusList"]) {
+      this.graph.current.highlightFocuses([]);
```

Comment:
Don't call this here; instead, just `setState` here, and pass that state as a prop to the `Graph` component and have that component call `highlightFocuses`.